context/FarmContext/index.tsx

```diff
@@ -310,41 +310,57 @@ export const FarmStore: React.FC<
         [signer, provider, config, account],
     );
 
-    const refreshTcrPriceUSDC = async () => {
-        if (!config?.sushiRouterAddress || !config?.tcrAddress || !config.usdcAddress || !signer) {
-            // leave it as the default value
+    const refreshRewardsTokenPriceUSDC = async ({ address, decimals }: { address?: string; decimals?: number }) => {
+        if (!config?.sushiRouterAddress || !address || !decimals || !config.usdcAddress || !signer) {
+            // no update to perform
             return;
         }
+
         const sushiRouter = new ethers.Contract(
             config?.sushiRouterAddress,
             UniswapV2Router02__factory.abi,
             signer,
         ) as UniswapV2Router02;
 
-        const oneTcr = new BigNumber('1').times(10 ** TCR_DECIMALS);
+        const oneUnit = new BigNumber('1').times(10 ** decimals);
 
-        const [usdcPer1Tcr] = await sushiRouter.getAmountsIn(oneTcr.toFixed(), [config.usdcAddress, config.tcrAddress]);
+        const [, buyPrice] = await sushiRouter.getAmountsOut(oneUnit.toFixed(), [address, config.usdcAddress]);
```

Comment:
is the comma here intentional

Reply:
yep, we want the second element of the array now that we are using `getAmountsOut` (gives the price you get for selling instead of the price you pay to buy one - we never really noticed with TCR but for FXS the spread is significant, $44 vs $33)

Reply:
oof. Yep lgtm was just checking